rust/lance-core/src/utils/mask.rs

```diff
@@ -265,12 +266,17 @@ impl std::ops::BitOr for RowIdMask {
 
 /// A collection of row ids.
 ///
+/// These row ids may either be stable-style (where they can be an incrementing
+/// u64 sequence) or address style, where they are a fragment id and a row offset.
+/// When address style, this supports setting entire fragments as selected,
+/// without needing to enumerate all the ids in the fragment.
+///
```

Comment:
Took me a minute to figure this out.  Even if these are stable-style we store them in the same nested 32-bit map structure? (though we will have far fewer entries in the outer map)

Reply:
Yes. To be fair, the roaring crate has a RoaringTreeMap it uses to store u64 values, so it’s not all different from that. The vast majority of tables will just use a single RoaringBitmap here, since most won’t have over 4 billions rows I expect.